c++/src/kj/compat/http.c++

```diff
@@ -2309,9 +2297,10 @@ public:
   }
 
   kj::Promise<void> flush() {
-    auto fork = writeQueue.fork();
-    writeQueue = fork.addBranch();
-    return fork.addBranch();
+    KJ_IF_MAYBE(promise, writeQueue) {
+      co_await *promise;
```

Comment:
If awaiting `promise` throws, then `writeQueue` won't be nulled out, but will contain a moved-away promise, so the next use will crash.

Perhaps we should define `flush` as:

```
kj::Promise<void> flush() {
  KJ_IF_MAYBE(promise, writeQueue) {
    auto result = kj::mv(*promise);
    writeQueue = nullptr;
    return result;
  } else {
    return kj::READY_NOW;
  }
}
```

This approach also avoids allocating a coroutine frame for every call, so should be more efficient.